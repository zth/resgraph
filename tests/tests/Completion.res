// @g
//  ^com

// @gql.ty type someType = {one: int}
//        ^com

// let _ = (~ct)
//             ^com

// @gql.field let me = (_: query, ~c) => {None}
//                                  ^com

type x
