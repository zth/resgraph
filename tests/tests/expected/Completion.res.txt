Complete tests/Completion.res 0:4
Attribute id:g:[0:3->0:5] label:g
{"status": "Completion", "items": [{
    "label": "gql.type",
    "kind": 4,
    "tags": [],
    "detail": "Indicates that the annotated record is a GraphQL Object Type.",
    "documentation": null
  }, {
    "label": "gql.interface",
    "kind": 4,
    "tags": [],
    "detail": "Indicates that the annotated record is a GraphQL interface.",
    "documentation": null
  }, {
    "label": "gql.interfaceResolver",
    "kind": 4,
    "tags": [],
    "detail": "",
    "documentation": null
  }, {
    "label": "gql.field",
    "kind": 4,
    "tags": [],
    "detail": "",
    "documentation": null
  }, {
    "label": "gql.enum",
    "kind": 4,
    "tags": [],
    "detail": "",
    "documentation": null
  }, {
    "label": "gql.union",
    "kind": 4,
    "tags": [],
    "detail": "",
    "documentation": null
  }, {
    "label": "gql.inputObject",
    "kind": 4,
    "tags": [],
    "detail": "",
    "documentation": null
  }, {
    "label": "gql.scalar",
    "kind": 4,
    "tags": [],
    "detail": "",
    "documentation": null
  }, {
    "label": "gql.type snippet - connection",
    "kind": 4,
    "tags": [],
    "detail": "Boilerplate for creating a new GraphQL connection for pagination.",
    "documentation": null,
    "sortText": "a",
    "insertText": "gql.type\n/** An edge in a connection. */\ntype ${1:entity}Edge = {\n  /** A cursor for use in pagination. */\n  @gql.field\n  cursor: string,\n  /** The item at the end of the edge. */\n  @gql.field\n  node: option<${1:entity}>\n}\n\n/** A connection to a list of items. */\n@gql.type\ntype ${1:entity}Connection = {\n  /** Information to aid in pagination. */\n  @gql.field\n  pageInfo: ResGraph.Connections.pageInfo,\n  /** A list of edges. */\n  @gql.field\n  edges: option<array<option<${1:entity}Edge>>>\n}",
    "insertTextFormat": 2
  }]}
Complete tests/Completion.res 3:10
Attribute id:gql.ty:[3:3->3:10] label:gql.ty
{"status": "Completion", "items": [{
    "label": "gql.type",
    "kind": 4,
    "tags": [],
    "detail": "Indicates that the annotated record is a GraphQL Object Type.",
    "documentation": null
  }, {
    "label": "gql.type snippet - connection",
    "kind": 4,
    "tags": [],
    "detail": "Boilerplate for creating a new GraphQL connection for pagination.",
    "documentation": null,
    "sortText": "a",
    "insertText": "gql.type\n/** An edge in a connection. */\ntype ${1:entity}Edge = {\n  /** A cursor for use in pagination. */\n  @gql.field\n  cursor: string,\n  /** The item at the end of the edge. */\n  @gql.field\n  node: option<${1:entity}>\n}\n\n/** A connection to a list of items. */\n@gql.type\ntype ${1:entity}Connection = {\n  /** Information to aid in pagination. */\n  @gql.field\n  pageInfo: ResGraph.Connections.pageInfo,\n  /** A list of edges. */\n  @gql.field\n  edges: option<array<option<${1:entity}Edge>>>\n}",
    "insertTextFormat": 2
  }]}
